feat(Button): iTwinUI-CSS 1.0.0 refactor

gretanausedaite opened the pull request:
Adding changes made in [#684](https://github.com/iTwin/iTwinUI/pull/684)

### Button variants & types

- `<button class="iui-button iui-default">` -> `<button class="iui-button">`
- `<button class="iui-button iui-high-visibility">` -> `<button class="iui-button" data-iui-variant="high-visibility">`
- `<button class="iui-button iui-cta">` -> `<button class="iui-button" data-iui-variant="cta">`
- `<button class="iui-button iui-borderless">` -> `<button class="iui-button" data-iui-variant="borderless">`
- `<button class="iui-button iui-idea">` -> `<button class="iui-button" data-iui-variant="idea">`
- `<button class="iui-button iui-dropdown">` -> `<button class="iui-button iui-button-dropdown">`

### Button sizes

- `<button class="iui-button iui-small">` -> `<button class="iui-button" data-iui-size="small">`
- `<button class="iui-button iui-large">` -> `<button class="iui-button" data-iui-size="large">`

### Active button state

- `<button class="iui-button iui-active">` -> `<button class="iui-button" data-iui-active="true">`


### Disabled button states !! Not sure how to add this !!

```diff
<button disabled>
<button disabled="disabled">
+ <button aria-disabled="true">
```

### Button label

- `<span class="iui-button-label">Hello</span>` -> `<span>Hello</span>"`

### Split menu

- `<span class="iui-button-split-menu">` -> `<span class="iui-button-split">`

bentleyvk committed:
fix(Table): Prevents crashing when clearing editable cell (#732)

bentleyvk committed:
fix(Table): Prevents crashing when clearing editable cell (#732)

olivertembo committed:
feat(ErrorPage): Add time-out and redirect status codes (#715)

olivertembo committed:
feat(ErrorPage): Add time-out and redirect status codes (#715)

gretanausedaite committed:
fix(Combobox): Allow users to reset value (#728)

Allow users to set value as undefined.

gretanausedaite committed:
fix(Combobox): Allow users to reset value (#728)

Allow users to set value as undefined.

github-actions[bot] committed:
chore: release 1.41.0 (#729)

github-actions[bot] committed:
chore: release 1.41.0 (#729)

mayank99 committed:
fix(Small): Use `small` element instead of `p` (#735)

mayank99 committed:
fix(Small): Use `small` element instead of `p` (#735)

bentleyvk committed:
chore: Updated glob-parent (#739)

bentleyvk committed:
chore: Updated glob-parent (#739)

dependabot[bot] committed:
chore(deps): bump terser from 4.8.0 to 4.8.1 (#741)

dependabot[bot] committed:
chore(deps): bump terser from 4.8.0 to 4.8.1 (#741)

bentleyvk committed:
chore: Updated `terser` package (#742)

bentleyvk committed:
chore: Updated `terser` package (#742)

r100-stack committed:
story(Slider): Swapped min/max labels for custom thumb slider (#743)

* Swapped the min and max labels for the slider with the custom thumb

* Approved test image

r100-stack committed:
story(Slider): Swapped min/max labels for custom thumb slider (#743)

* Swapped the min and max labels for the slider with the custom thumb

* Approved test image

gretanausedaite committed:
fix(ComboBox): Prevent infinite loop when `options` change (#738)

- Added ref to `options` prop;
- Added ref to `valueProp`;

gretanausedaite committed:
fix(ComboBox): Prevent infinite loop when `options` change (#738)

- Added ref to `options` prop;
- Added ref to `valueProp`;

HaveSpacesuit committed:
feat(Table): Scroll table to selected item (#689)

Co-authored-by: Vilius <36186912+bentleyvk@users.noreply.github.com>

Commit:
Making changes for css v1

Commit:
Making changes for css v1

veekeys committed:
feat(Tree): Virtualization (#713)

github-actions[bot] committed:
chore: release 1.42.0 (#751)

Commit:
fixing tests

Commit:
fixing tests

Commit:
Undoing 'main' changes

Commit:
Undoing 'main' changes

Commit:
Undoing 'main' changes

Commit:
Merge branch 'greta/button-v1-refactor' of https://github.com/iTwin/iTwinUI-react into greta/button-v1-refactor

mayank99 commented on packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx:
```diff
@@ -71,14 +71,9 @@ export const Button: ButtonComponent = React.forwardRef((props, ref) => {
-        className,
-      )}
+      className={cx('iui-button', className)}
+      data-iui-variant={styleType}
+      data-iui-size={size || undefined}
```
I think we don't want to add a data attribute for default variant and size, so we might need a ternary conditional here

gretanausedaite replied:
Size doesn't have `default` value. It's small or large. 
![image](https://user-images.githubusercontent.com/83585998/181451978-f9fe6037-49fa-470b-b02c-fbdc4b032261.png)

Added condition to variant data attribute.

mayank99 commented on packages/iTwinUI-react/src/core/Buttons/IconButton/IconButton.tsx:
```diff
@@ -46,15 +46,10 @@ export const IconButton: IconButtonComponent = React.forwardRef(
-          className,
-        )}
+        className={cx('iui-button', className)}
+        data-iui-variant={styleType}
+        data-iui-size={size || ''}
```
Same comment here

mayank99 commented on packages/iTwinUI-react/src/core/Alert/Alert.tsx:
```diff
@@ -96,14 +90,14 @@ export const Alert = (props: AlertProps) => {
         >
-          <SvgCloseSmall aria-hidden />
-        </IconButton>
+          <SvgCloseSmall aria-hidden className='iui-alert-button-icon' />
```
Why did this file change? Shouldn't it be handled in #753?

gretanausedaite replied:
It is handled there. I don't know why github sees this change. It does not show up in VS code PR 
![image](https://user-images.githubusercontent.com/83585998/181429855-6c150924-f85d-4c38-8e74-0b258ff340c1.png)

veekeys replied:
I'd say branch is missing a merge with v1

gretanausedaite replied:
Fixed

Commit:
Condition to exclude default variant

adamhock committed:
feat(Table): itwinui-css 1.0.0 Class Refactor (#749)

### **Changes**
- **Data Attribute**
  - `iui-condensed/extra-condensed` replaced with `data-iui-size="condensed/extra-condensed"`
  - `iui-selected` replaced with `aria-selected="true"`
    - `iui-table-body` elements that contain selectable rows now have `aria-multiselectable="true"`
  - `iui-disabled` replaced with `aria-disabled="true"`
- **Class name changes**
  - `iui-paginator-*` changed to `iui-table-paginator-*`
  - `iui-column-filter` changed to `iui-table-column-filter`
  - `iui-filter-button` changed to `iui-table-filter-button`
  - `iui-resizer-*` changed to `iui-table-resizer-*`
  - `iui-sort` changed to `iui-table-sort`
  - `iui-reorder-bar-*` changed to `iui-table-reorder-bar-*`
  - `iui-expanded-content` changed to `iui-table-expanded-content`
  - `iui-row-*` changed to `iui-table-row-*`
  - `iui-cell-*` changed to `iui-table-cell-*`

Commit:
merge with v1

Commit:
merge with v1 2

Commit:
adding random change

veekeys committed:
Merge branch 'itwinui-css-v1' of https://github.com/iTwin/iTwinUI-react into greta/button-v1-refactor

Commit:
Merge branch 'itwinui-css-v1' of https://github.com/iTwin/iTwinUI-react into greta/button-v1-refactor

Commit:
Merge branch 'greta/button-v1-refactor' of https://github.com/iTwin/iTwinUI-react into greta/button-v1-refactor

Commit:
undo random change

veekeys commented on packages/iTwinUI-react/src/core/Buttons/Button/Button.tsx:
```diff
@@ -71,14 +71,11 @@ export const Button: ButtonComponent = React.forwardRef((props, ref) => {
-      )}
+      className={cx('iui-button', className)}
+      data-iui-variant={
+        styleType && styleType !== 'default' ? styleType : undefined
```
`styleType` is always defined, so enough
`styleType !== `default` ? styleType : undefined`

gretanausedaite replied:
updated

veekeys commented on packages/iTwinUI-react/src/core/Buttons/IconButton/IconButton.tsx:
```diff
@@ -46,15 +46,12 @@ export const IconButton: IconButtonComponent = React.forwardRef(
+          styleType && styleType !== 'default' ? styleType : undefined
+        }
+        data-iui-size={size}
+        data-iui-active={isActive}
```
hmm... Interesting.. in comment, we said `isActive default value is false`, but we do not assign it in props destruction...
But it makes sense not to have attribute with `false` value, so lets leave it, I guess...

Commit:
Removing styleType defined check
